### PR TITLE
[TRIVIAL] Resolve compiler warning

### PR DESCRIPTION
**Pull request by seelabs**

### CODE REVIEW

**Comment by seelabs** [comment]
We could be fancy and define our own assert macro that would use the variables in a `sizeof` to resolve the warning (from stack overflow):```
#ifdef NDEBUG
#define ASSERT(x) do { (void)sizeof(x);} while (0)
#else
#include <assert.h>
#define ASSERT(x) assert(x)
#endif```
However unless it becomes a bigger problem I think the simple `(void)x` to resolve this is best.

File: src/ripple/app/ledger/impl/LedgerMaster.cpp [hunk 1]
```diff
@@ -211,6 +211,7 @@ LedgerMaster::setValidLedger(
             !app_.getMaxDisallowedLedger() ||
             l->info().seq + max_ledger_difference_ >
                     app_.getMaxDisallowedLedger());
+    (void) max_ledger_difference_;
```
Comment (miguelportilla) [inline src/ripple/app/ledger/impl/LedgerMaster.cpp]: Not necessary but it might be helpful to add a comment describing why this is here.

**Comment by miguelportilla** [comment]
The problem with the fancy solution is with release builds that pack asserts, some folks like to test with such builds.

File: src/ripple/app/ledger/impl/LedgerMaster.cpp (hunk 1 above)
Comment (seelabs) [inline src/ripple/app/ledger/impl/LedgerMaster.cpp]: I think we do that enough we don't need a comment. `(void)some_var` is clearly to resolve a warning. I'd rather not draw attention to something so unimportant with a comment.

**Comment by nbougalis** [comment]
Merged as 5c1ec051f0dbc4cfcb76addb836354ea4710b076

**Pull request closed without merging**